core/src/main/java/io/kestra/core/models/tasks/runners/TaskRunner.java

```
@@ -101,4 +113,22 @@ public String toAbsolutePath(RunContext runContext, TaskCommands taskCommands, S
 
         return workingDir + "/" + relativePath;
     }
+    
+    /** {@inheritDoc} **/
+    @Override
+    public void kill() {
+        if (isKilled.compareAndSet(false, true)) {
```

I don't think it's ever possible to call two times kill() on the same object as serialization will happens just after killing so the field will be set back to it's default.

---

But are we sure that two Killed events cannot be sent for the same execution (e.g. two fast and successive calls to REST API)?  It's not very expensive to have a defensive programming approach. 

In addition, I wondered what would happen if a kill event was handled by the task before a job/container was actually created (i.e. we're in the middle of the run method). In this context, I think there's a good chance that the task will be executed, unless the task's thread interrupt flag is always set to true when a blocking method is invoked (but this behavior is really implicit).

It might be better to raise an exception when registering the kill runnable if the task has been killed in the meantime. WDYT ? @loicmathieu

---

You are right, you can receive multiple kill event.

> It might be better to raise an exception when registering the kill runnable if the task has been killed in the meantime.

> In addition, I wondered what would happen if a kill event was handled by the task before a job/container was actually created

The thread will be interrupted so nothing will be created, you can test that if you"re curious ;)